test/compress/classes.js

```diff
@@ -0,0 +1,574 @@
+                yield A.#p * 3;
+            }
+            async #g() {
+                for (var a of A.#f())
+                    return a * await 2;
+            }
+            static get #p() {
+                return 7;
+            }
+            get q() {
+                return this.#g();
+            }
+        }().q.then(console.log);
+    }
+    expect_exact: "(new class A{static*#f(){yield 3*A.#p}async #g(){for(var a of A.#f())return a*await 2}static get #p(){return 7}get q(){return this.#g()}}).q.then(console.log);"
+    expect_stdout: "42"
+    node_version: ">=14"
+}
+
+await: {
+    input: {
+        var await = "PASS";
+        (async function() {
+            return await new class extends (await function() {}) { [await 42] = await };
+        })().then(function(o) {
+            console.log(o[42]);
+        });
+    }
+    expect_exact: 'var await="PASS";(async function(){return await new class extends(await function(){}){[await 42]=await}})().then(function(o){console.log(o[42])});'
+    expect_stdout: "PASS"
+    node_version: ">=12"
+}
+
+yield: {
+    input: {
+        var a = function*() {
+            yield new class { [yield "foo"] = "bar" };
+        }();
+        console.log(a.next().value);
+        console.log(a.next(42).value[42]);
+    }
+    expect_exact: 'var a=function*(){yield new class{[yield"foo"]="bar"}}();console.log(a.next().value);console.log(a.next(42).value[42]);'
+    expect_stdout: [
+        "foo",
+        "bar",
+    ]
+    node_version: ">=12"
+}
+
+conditional_parenthesis: {
+    options = {
+        conditionals: true,
+    }
+    input: {
+        "use strict";
+        if (class {})
+            console.log("PASS");
+    }
+    expect_exact: '"use strict";(class{})&&console.log("PASS");'
+    expect_stdout: "PASS"
+    node_version: ">=4"
+}
+
+class_super: {
+    input: {
+        "use strict";
+        class A {
+            static get p() {
+                return "Foo";
+            }
+            static get q() {
+                return super.p || 42;
+            }
+            constructor() {
+                console.log("a.p", super.p, this.p);
+                console.log("a.q", super.q, this.q);
+            }
+            get p() {
+                return "foo";
+            }
+            get q() {
+                return super.p || null;
+            }
+        }
+        class B extends A {
+            static get p() {
+                return "Bar";
+            }
+            static get q() {
+                return super.p;
+            }
+            constructor() {
+                super();
+                console.log("b.p", super.p, this.p);
+                console.log("b.q", super.q, this.q);
+            }
+            get p() {
+                return "bar";
+            }
+            get q() {
+                return super.p;
+            }
+        }
+        console.log("A", A.p, A.q);
+        console.log("B", B.p, B.q);
+        new B();
+    }
```

Comment:
If you replace those getters with fields, the result will surprise ye~

Reply:
I tried but didn't see anything unusual. Do you mean that static properties are inherited from the base?

Reply:
Original output:
```
A Foo 42
B Bar Foo
a.p undefined bar
a.q undefined foo
b.p foo bar
b.q null foo
```
<details>
<summary>Replace getters with fields:</summary>

```js
"use strict";
class A {
    static p = "Foo";
    static q = super.p || 42;
    constructor() {
        console.log("a.p", super.p, this.p);
        console.log("a.q", super.q, this.q);
    }
    p = "foo";
    q = super.p || null;
}
class B extends A {
    static p = "Bar";
    static q = super.p;
    constructor() {
        super();
        console.log("b.p", super.p, this.p);
        console.log("b.q", super.q, this.q);
    }
    p = "bar";
    q = super.p;
}
console.log("A", A.p, A.q);
console.log("B", B.p, B.q);
new B();
```

</details>

Replaced output:
```
A Foo 42
B Bar Foo
a.p undefined foo
a.q undefined null
b.p undefined bar
b.q undefined undefined
```

Reply:
Ah those class fields. I don't think they're in the ES standard yet... https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Public_class_fields, although most browsers and tools (rollup, esbuild, babel) implement some subset of the proposed features.

I think the way class instance fields work is prepending the initialisations to the constructor.  I don't know how the class static ones are supposed to be initialized. Different implementations do not agree on the use of `super` in field initialisations.

Reply:
In the ideal world you'd hope they make it work the same way as in object literals &minus; not that the latter is working intuitively IMO